cmd/ctl/pkg/status/certificate/certificate.go

```diff
@@ -0,0 +1,169 @@
+	// The Namespace that the Certificate to be renewed resided in.
+	// This flag registration is handled by cmdutil.Factory
+	Namespace string
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdStatusCert returns a cobra command for status certificate
+func NewCmdStatusCert(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+	cmd := &cobra.Command{
+		Use:     "certificate",
+		Short:   "Get details about the current status of a Certificate",
+		Long:    long,
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Validate(args))
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(args []string) error {
+	if len(args) < 1 {
+		return errors.New("the name of the Certificate has to be provided as argument")
+	}
+	if len(args) > 1 {
+		return errors.New("only one argument can be passed in: the name of the Certificate")
+	}
+	return nil
+}
+
+// Complete takes the factory and infers any remaining options.
+func (o *Options) Complete(f cmdutil.Factory) error {
+	var err error
+
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.RESTConfig, err = f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.CMClient, err = cmclient.NewForConfig(o.RESTConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes status certificate command
+func (o *Options) Run(args []string) error {
+	ctx := context.TODO()
+	crtName := args[0]
+
+	crt, err := o.CMClient.CertmanagerV1alpha2().Certificates(o.Namespace).Get(ctx, crtName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error when getting Certificate resource: %v", err)
+	}
+
+	fmt.Fprintf(o.Out, fmt.Sprintf("Name: %s\nNamespace: %s\n", crt.Name, crt.Namespace))
+
+	// Get necessary info from Certificate
+	// Output one line about each type of Condition that is set.
+	conditionMsg := ""
+	for i, con := range crt.Status.Conditions {
+		if i < len(crt.Status.Conditions)-1 {
+			conditionMsg += fmt.Sprintf("  %s: %s, Reason: %s, Message: %s\n", con.Type, con.Status, con.Reason, con.Message)
+		} else {
+			conditionMsg += fmt.Sprintf("  %s: %s, Reason: %s, Message: %s", con.Type, con.Status, con.Reason, con.Message)
+		}
+	}
+	if conditionMsg == "" {
+		conditionMsg = "  No Conditions set"
+	}
+	fmt.Fprintf(o.Out, fmt.Sprintf("Conditions:\n%s\n", conditionMsg))
+
+	// TODO: Get CR from certificate if exists. I think I can just look for it without caring what condition is
+	// What about timing issues? When I query condition it's not ready yet, but then looking for crn it's finished and deleted
+
+	dnsNames := formatDnsNamesList(crt)
```

Comment:
Perhaps this could be `formatStringSlice(crt.Spec.DNSNames)`? We can re-use it then any other time we want to print a list of strings 😄